src/features/CareViewPatients/components/CareViewPatients.jsx

```diff
@@ -36,46 +46,117 @@ export const CareViewPatients = () => {
     }
   };
 
-  useEffect(() => {
-    if (currentPage > 1) {
-      setCurrentPage(1);
+  const getPatientsListBySearch = async (offset = 1) => {
+    try {
+      setIsLoading(true);
+      const response = await fetchPatientsListBySearch(
+        selectedWard.value,
+        searchKeys,
+        searchValue,
+        (offset - 1) * limit,
+        limit
+      );
+      if (response.status === 200) {
+        const { admittedPatients, totalPatients } = response.data;
+        setPatientList(admittedPatients);
+        setTotalPatients(totalPatients);
+      } else {
+        throw new Error("Failed to fetch data");
+      }
+    } catch (error) {
+      console.error("Error fetching data:", error);
+    } finally {
+      setIsLoading(false);
     }
+  };
+
+  useEffect(() => {
+    setCurrentPage(1);
+    updateSearchValue("");
+    setIsSearched(false);
+    getPatientsList();
   }, [selectedWard]);
 
   useEffect(() => {
     if (selectedWard.value) {
-      getPatientsList();
+      if (isSearched) {
+        getPatientsListBySearch(currentPage);
+      } else {
+        getPatientsList();
+      }
     }
-  }, [selectedWard, currentPage, limit]);
+  }, [currentPage, limit]);
 
   const handlePaginationChange = (e) => {
     setCurrentPage(e.page);
     setLimit(e.pageSize);
   };
 
+  const handleKeyPress = (event) => {
+    if (event.key === "Enter") {
+      if (searchValue.length >= 3) {
+        if (!isSearched) {
+          setPreviousPage(currentPage);
+        }
+        setIsSearched(true);
+        var isCurrentPageOne = currentPage === 1;
```

Comment:
use const instead of "var" and please check for other file changes as well

Reply:
Sure